### PR TITLE
Integrate Spinnaker with sig-events PoC

### PR DESCRIPTION
**Pull request by rjalander**
This PR includes the changes as described below;

**poc.sh** - Verify whether spincli is installed as part of Prerequisites 
**installAndConfigSpinnaker.sh** - a script invoking from poc.sh to integrate spinnaker with the PoC and it does below actions;
1. Create hal stable deployment and update with Spinnaker service account
2. Deploy distributed spinnaker in K8S cluster using halyard
3. Update spin-echo with the cdevents API and build new image
4. Create a trigger to subscribe spinnaker event API
5. Launch spinnaker deployment and create Application/Pipeline using spin CLI
6. Create Ingress to access spinnaker UI from host

**deploySpinnakerWithHalyard.sh** - a script invoking from installAndConfigSpinnaker.sh to deploy spinnaker using halyard inside K8S cluster and it does below actions;
1. Install helm CLI 
2. Configure default K8S context inside halyard pod 
3. Install Minio Server as service in k8s cluster and configure as storage service for spinnaker
4. Configure kubernetes account with spinnaker
5. deploy spinnaker using halyard command

**deploy-spinnaker-poc.json** - a template used to create a pipeline in Spinnaker using slpincli

**spin-echo-deploy.yaml** -  to replace the spin-echo micro service to use the spinnaker consume event API from private branch as for now.
**spinnakerServiceAccount.yaml** - Creates spinnaker service account using ClusterRoleBinding to access spinnaker

### CODE REVIEW

**Review by afrittoli — COMMENTED** [review]
Thanks again for this @rjalander and sorry it took so long for me to look into it.

The main concern I have is that with this PR the old PoC is not available (on `main`) anymore.
I would ask you if possible to:

- either copy the original folder into a new one and make the changes there
- or add some flags to the PoC script so that it's possible to enable spinnaker, but it's not used by default - and add a new PNG diagram rather than altering the original one

It would be very interesting to the the PoC in action as well. Does any of the existing meeting work for you? If not, we're working on setting up an alternate time meeting, so that might be an option soon-ish

**Comment by rjalander** [comment]
Thank you for reviewing this PR @afrittoli,
I have made the changes as suggested, added flag (-s) to run the Spinnaker with CDEvents PoC
Now, `./poc.sh -s` Runs PoC with Spinnaker installation, `./poc.sh` goes with Keptn and Tekton as usual.

Sure, this demo part is discussed in the last sig-events meeting to set the date.
I am fine with any of the existing meeting or any other time that suits you.

**Comment by e-backmark-ericsson** [comment]
@rjalander , could we plan for a demo of this integration to one of the upcoming SIG Events / CDEvents meetings, to provide some background info and an easier entry intro into this review? It would be preferred to have it on one of the coming SIG meetings, on July 4th, July 19th or August 1st, but it could also be on one of the upcoming CDEvents meetings (July 12th or 26th). Which would you prefer?

**Comment by rjalander** [comment]
@e-backmark-ericsson , we can have it on today's (July 4th) SIG meeting If the time permits for demo and review the PR changes.

File: poc/spinnaker/deploy-spinnaker-poc.json [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+      "template": {
+       "metadata": {
+        "labels": {
+         "app": "cdevents-spinnaker-poc"
+        }
+       },
+       "spec": {
+        "containers": [
+         {
+          "image": "localhost:5000/cdevent/poc:v1.0",
```
[the remark this answers is not in the document]
Comment (rjalander) [reply poc/spinnaker/deploy-spinnaker-poc.json:37]: @e-backmark-ericsson @afrittoli I can able to get the latest sha256 image from artifact published data and deploy that in Spinnaker.
The code change is in Spinnaker code base to update the Pipeline with the latest `artifactid` before triggering the Pipeline build.

{3BDC1528-6713-478C-9CB6-AD9D3C03172E}

**Review by afrittoli — APPROVED** [review]
Thank you for the great demo and updates!

As a follow-up to this PR we will need to figure out why the Tekton deployment is triggered multiple times, which should not happen.

@rjalander Is it ok if I squash all the commits when merging?

**Comment by rjalander** [comment]
Sure @afrittoli , can be squashed.

Also when running this PoC a couple of weeks back with the main branch of `tektoncd/experimental` I see tekton was unable to emit the cloudevent and the below error comes in `tekton-cloudevents-controller` pod logs 

> {"level":"warn","ts":1649693382.4827335,"logger":"cloudevents-pipelinerun-controller","caller":"events/events.go:47","msg":"Failed to emit cloud events no cloud events client found in the context","knative.dev/controller":"github.com.tektoncd.experimental.cloudevents.pkg.reconciler.pipelinerun.Reconciler","knative.dev/kind":"tekton.dev.PipelineRun","knative.dev/traceid":"18e39654-d287-4f6f-99a8-6892a8eb7c6f","knative.dev/key":"default/build-artifact-run-sdlc5"}

I checked out the branch from this previous commit `git checkout 8c1b6041bc679299750e00e551a8b137efcc5d65 github.com/tektoncd/experimental`, just to unblock myself.

But, I am yet to test If this issue still exist with the master branch now.

**Pull request merged**